# Examining Racial Discrimination in the US Job Market

### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés to black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes when presented to the employer.

<div class="span5 alert alert-info">
<h3> Exercises </h3>
You will perform a statistical analysis to establish whether race has a significant impact on the rate of callbacks for resumes.

Answer the following questions **in this notebook below and submit to your Github account**. 

   <p>1. What test is appropriate for this problem? Does CLT apply?</p>
   <p>2. What are the null and alternate hypotheses?</p>
   <p>3. Compute margin of error, confidence interval, and p-value.</p>
   <p>4. Write a story describing the statistical significance in the context or the original problem.</p>
   <p>5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?</p>

You can include written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet


<p> Resources <p>
+ Experiment information and data source: http://www.povertyactionlab.org/evaluation/discrimination-job-market-united-states
+ Scipy statistical methods: http://docs.scipy.org/doc/scipy/reference/stats.html 
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet
</div>

In [1]:
import pandas as pd
import numpy as np
import math
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline
import scipy as sp
from scipy import stats

In [2]:
data = pd.io.stata.read_stata('E:\\Data Science\\Assignments\\racial_disc\\data\\us_job_market_discrimination.dta')

In [3]:
data.head()

id ad  education  ofjobs  yearsexp  honors  volunteer  military  empholes  \
0  b  1          4       2         6       0          0         0         1   
1  b  1          3       3         6       0          1         1         0   
2  b  1          4       1         6       0          0         0         0   
3  b  1          3       4         6       0          1         0         1   
4  b  1          3       3        22       0          0         0         0   

   occupspecific    ...      compreq  orgreq  manuf  transcom  bankreal trade  \
0             17    ...          1.0     0.0    1.0       0.0       0.0   0.0   
1            316    ...          1.0     0.0    1.0       0.0       0.0   0.0   
2             19    ...          1.0     0.0    1.0       0.0       0.0   0.0   
3            313    ...          1.0     0.0    1.0       0.0       0.0   0.0   
4            313    ...          1.0     1.0    0.0       0.0       0.0   0.0   

  busservice othservice  missind  ownership  
0        0.0        0.0      0.0             
1        0.0        0.0      0.0             
2        0.0        0.0      0.0             
3        0.0        0.0      0.0             
4        0.0        1.0      0.0  Nonprofit  

[5 rows x 65 columns]

In [4]:
# number of callbacks for black-sounding names
sum(data[data.race=='b'].call)

157.0

In [5]:
sum(data[data.race=='w'].call)

235.0

In [6]:
data.shape

(4870, 65)

In [7]:
data[['race','call']].head()

race  call
0    w   0.0
1    w   0.0
2    b   0.0
3    b   0.0
4    w   0.0

In [8]:
data_white = data[data.race=='w']

In [9]:
data_black = data[data.race=='b']

In [10]:
data_white.call.mean(), data_black.call.mean()

(0.09650924056768417, 0.0644763857126236)

In [11]:
data_white.call.var(), data_black.call.var()

(0.08722896873950958, 0.060343656688928604)

In [12]:
data_white.call.std(), data_black.call.std()

(0.2953455150127411, 0.24564945697784424)

In [13]:
diff_mean = data_white.call.mean()-data_black.call.mean()
diff_mean

0.03203285485506058

In [14]:
len(data_white.call), len(data_black.call)

(2435, 2435)

# 1. What test is appropriate for this problem? Does CLT apply?

The sample contains data values > 30. Also we are not aware of the population standard deviation. **t-test** is advisable for this problem.

This is a binomial distribution i.e. the candidates either get a call back or no. For every large sample, a binomial distribution can be approximated to normal distrbution. This being a fairly large sample (4870), the Central Limit Theorem applies in this case. 

# 2. What are the null and alternate hypotheses?

Hypothesis defined as:

- $H$o: White and black resumes get same responses i.e. **x̄ - ȳ = 0**

- $H$a: White and black resumes do not get same responses  i.e **x̄ - ȳ ≠ 0**

# 3. Compute margin of error, confidence interval, and p-value.

**Margin of Error = (t_critical) * (standard error)**

t_critical value for 4877 Degrees of freedom with a significane level of 0.05 = 1.962

**SE(w-b) = √ ($s_1^2/n_1) + (s_2^2/n_2$)**


In [15]:
standard_error = math.sqrt(((data_white.call.var()/len(data_white.call))+(data_black.call.var()/len(data_black.call))))
standard_error

0.007784906826883674

In [16]:
MOE = 1.962*standard_error
MOE

0.015273987194345768

Since the data is assumed to be normally distributed, let us calculate a 95% confidence interval for the disribution.

The Confidence Interval for the distribution (difference of black and white) lies within +/- 0.01527 of the difference of the means i.e. 0.0320 (diff_mean)

**Confidence Interval - (0.0167,0.0472)**

** p-value: 0.00003940**

In [17]:
sp.stats.ttest_ind(data_white.call, data_black.call)

Ttest_indResult(statistic=4.1147052908617514, pvalue=3.9408021031288859e-05)

# 4. Write a story describing the statistical significance in the context or the original problem.

Having assumed that the null hypothesis is true (White and black resumes get same responses) there is a 5% chance that difference between the two sample means (black calls and white calls) to be more than 0.015. But the difference in the mean we obtained is 0.032 which is way above 0.015. Therefore we can reject the Null Hypothesis.

# 5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

From the statistics above there is a 9.65% chance that white-sounding name resumes hear back from recruiters compared to the 6.44% chance with black-sounding names.

The analysis above says that white sounding names recieve more calls than black sounding names, but the data is not sufficient enough to say if race plays an important factor on call back success. Candidates might not have recieved calls due to their qualifications or any other reason. 

The data provided is not sufficient enough to tell if race is the primary factor for the call black success. 